### PR TITLE
Remove IFLAGS/FFLAGS types

### PR DESCRIPTION
**Pull request by uweigand**
All instructions using the CPU flags types (IFLAGS/FFLAGS) were already removed.  This patch completes the cleanup by removing all remaining instructions that define values of CPU flags types, as well as the types themselves.

Specifically, the following features are removed:
- The IFLAGS and FFLAGS types and the SpecialType category.
- Special handling of IFLAGS and FFLAGS in machinst/isle.rs and machinst/lower.rs.
- The ifcmp, ifcmp_imm, ffcmp, iadd_ifcin, iadd_ifcout, iadd_ifcarry, isub_ifbin, isub_ifbout, and isub_ifborrow instructions.
- The writes_cpu_flags instruction property.
- The flags verifier pass.
- Flags handling in the interpreter.

All of these features are currently unused; no functional change intended by this patch.

This addresses https://github.com/bytecodealliance/wasmtime/issues/3249.

CC @cfallin @elliottt @jameysharp

### CODE REVIEW

File: cranelift/codegen/src/ir/dfg.rs [hunk 1]
```diff
@@ -1508,39 +1507,4 @@ mod tests {
         assert_eq!(dfg.value_is_attached(arg3), false);
         assert_eq!(dfg.block_params(block), &[]);
     }
-
-    #[test]
-    fn aliases() {
```
Comment (bjorn3) [inline cranelift/codegen/src/ir/dfg.rs:1513]: Rather than removing this test, could you change it to avoid iflags?

**Review by cfallin — APPROVED** [review]
Thanks a bunch for doing this work! It looks pretty straightforwardly correct -- just a bunch of special cases going away, which is always great to see.

Happy to merge once @bjorn3's suggestions are done!

File: cranelift/filetests/filetests/parser/ternary.clif [hunk 2]
```diff
@@ -1,24 +0,0 @@
...
-
-function %add_i96(i32, i32, i32, i32, i32, i32) -> i32, i32, i32 {
-block1(v1: i32, v2: i32, v3: i32, v4: i32, v5: i32, v6: i32):
-    v10, v11 = iadd_ifcout v1, v4
-    ;check: v10, v11 = iadd_ifcout v1, v4
-    v20, v21 = iadd_ifcarry v2, v5, v11
-    ; check: v20, v21 = iadd_ifcarry v2, v5, v11
-    v30 = iadd_ifcin v3, v6, v21
-    ; check: v30 = iadd_ifcin v3, v6, v21
-    return v10, v20, v30
```
[the remark this answers is not in the document]
Comment (uweigand) [reply cranelift/filetests/filetests/parser/ternary.clif:12]: Changed to `iadd_cout` and friends as well.  Also added a `select` test for completeness.

**Comment by cfallin** [comment]
CI breakage looks to be because of an Ubuntu upgrade; created #5407, and we'll want to rebase this on that if it works and merges.

**Comment by uweigand** [comment]
> CI breakage looks to be because of an Ubuntu upgrade; created #5407, and we'll want to rebase this on that if it works and merges.

Thanks!  Tests are passing now.

**Pull request merged**